### PR TITLE
Add org.octave.Octave_64_64.

### PR DESCRIPTION
**Pull request by siko1056**
This application "org.octave.Octave_64_64" is a demanded variant of GNU Octave https://github.com/flathub/org.octave.Octave, which enables to treat matrices beyond 32 GB of main memory.

For MS Windows, the GNU Octave project creates a similar set of variants (https://www.gnu.org/software/octave/download.html)

- octave-5.2.0_1-w64-installer.exe
- octave-5.2.0_1-w64-64-installer.exe

The idea is that I synchronize this new application with https://github.com/flathub/org.octave.Octave (maintained by https://github.com/mtmiller).  Only the app ID and the configuration are different.

Thank you for your time and help.

### CODE REVIEW

**Comment by barthalion** [comment]
bot, build org.octave.Octave_64_64

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "shared-modules"]
```
Comment (barthalion) [inline .gitmodules]: Seems this isn't proper git submodule. Mind re-adding it with `git submodule add`?

Comment (siko1056) [reply .gitmodules]: Thanks, I'll fix this.

**Comment by barthalion** [comment]
Can you patch the appdata file to mention in the description what the difference is between 64 variant and regular app? You should probably mention "64" in app name too.

cc @mtmiller

**Comment by siko1056** [comment]
@mtmiller I assume, that the file [`/etc/icons/org.octave.Octave.appdata.xml`](https://hg.savannah.gnu.org/hgweb/octave/file/eb46a9f47164/etc/icons/org.octave.Octave.appdata.xml) contained in the Octave release tarball is used.  Is this obvious from the `org.octave.Octave_64_64.yaml` file somehow?  Then I can add a patch.

**Comment by nedrichards** [comment]
So, the reason we wouldn't want to offer this by default is something like:
>  Unless your computer has more than ~32GB of memory and you need to solve linear algebra problems with arrays containing more than ~2 billion elements, this version will offer no advantage over the recommended Windows-64 version above. 

Are there any other downsides? Out of interest.

**Comment by siko1056** [comment]
> Are there any other downsides? Out of interest.

The major "downside" to my knowledge is, that these types of OpenBLAS, SuiteSparse, ... builds are not default by today.  Thus these configurations are less tested than the 32-bit-Integer versions.  But today's PCs are getting over this limit these days, thus it should get more tests/attention by a wider audience before becoming default.

**Comment by mtmiller** [comment]
Patch to the appdata description looks fine. The "name" field of the appdata and desktop entry should also mention the "64" variant. However, changing the app ID has further implications, you probably also want to rename both the appdata and the .desktop file (I think flatpak-builder can do that for you). And also patch libgui/src/main-window.cc [here](https://github.com/mtmiller/octave/blob/master/libgui/src/main-window.cc#L189).

I'm personally not convinced it's worth the effort to maintain these two different variants, but @siko1056 please pursue it if it's worth it to you, and the Flathub admins agree in general with the idea of publishing build flavors of applications (feels a little too Gentoo-ish to me).

Nitpick: I would prefer an app ID like "org.octave.Octave-BLAS64" (hyphen looks better than underscore, and BLAS64 clarifies the reason for the difference).

**Comment by siko1056** [comment]
@mtmiller Thanks for reviewing the patch.  The app ID "org.octave.Octave_BLAS_64" seems like a nice compromise.  Because of a [DBus limitation](http://docs.flatpak.org/en/latest/conventions.html#application-ids), hyphens are not allowed.

I open a new PR with the modified name.  Sorry for the noise.

**Pull request closed without merging**